newIDE/app/src/Profile/CurrentUsageDisplayer.js

```diff
@@ -23,29 +27,82 @@ const styles = {
   diamondIcon: {
     width: 50,
     height: 50,
+    // Prevent cumulative layout shift by enforcing the ratio.
+    aspectRatio: '1',
   },
 };
 
 type Props = {|
   subscription: ?Subscription,
   quota: ?Quota,
-  numberOfPendingBuilds: number,
+  usagePrice: ?UsagePrice,
   onChangeSubscription: () => void,
+  onStartBuildWithCredits: () => void,
+  hidePurchaseWithCredits?: boolean,
 |};
 
 const CurrentUsageDisplayer = ({
   subscription,
   quota,
-  numberOfPendingBuilds,
+  usagePrice,
   onChangeSubscription,
+  onStartBuildWithCredits,
+  hidePurchaseWithCredits,
 }: Props) => {
   const gdevelopTheme = React.useContext(GDevelopThemeContext);
-  if (!quota || !subscription) return <PlaceholderLoader />;
+  const { openCreditsPackageDialog, openCreditsUsageDialog } = React.useContext(
+    CreditsPackageStoreContext
+  );
+  const { profile, limits } = React.useContext(AuthenticatedUserContext);
+  const usageCreditPrice = usagePrice ? usagePrice.priceInCredits : 0;
+
+  const onPurchaseBuildWithCredits = React.useCallback(
+    async () => {
+      if (!profile || !limits || !usageCreditPrice) {
+        return;
+      }
+
+      const currentCreditsAmount = limits.credits.userBalance.amount;
+      if (currentCreditsAmount < usageCreditPrice) {
+        openCreditsPackageDialog({
+          missingCredits: usageCreditPrice - currentCreditsAmount,
+        });
+        return;
+      }
+
+      openCreditsUsageDialog({
+        title: <Trans>Start build with credits</Trans>,
+        message: (
+          <Trans>
+            You are about to use {usageCreditPrice} credits to start this build.
+            Continue?
+          </Trans>
+        ),
+        onConfirm: async () => {
+          // We do not await for the build to start, we assume
+          // that the ExportLauncher will handle the error if the build fails.
+          onStartBuildWithCredits();
+        },
+        successMessage: <Trans>🎉 Build started!</Trans>,
```

Comment:
We don't use emojis that much at the moment, should we?

Reply:
Not sure, I've used it in a few success messages, but agree it's not ideal. Anyway, this message is not visible as it's closed automatically, so I'll remove it